src/wp-includes/kses.php

```diff
@@ -2426,6 +2428,10 @@ function safecss_filter_attr( $css, $deprecated = '' ) {
 			$parts        = explode( ':', $css_item, 2 );
 			$css_selector = trim( $parts[0] );
 
+			if ( in_array( '--*', $allowed_attr, true ) && 0 === strpos( $css_selector, '--' ) && sanitize_key( $css_selector ) === $css_selector ) {
```

Comment:
Would it be an issue that `sanitize_key` forces lower case, so `--HOORAH` becomes `--hoorah`?

Also `in_array( '--*', $allowed_attr, true )` is always going to be `true`, no? 

What if we checked for `--` at the start of the `$css_selector`? I was playing with [something like this over in this PR](https://github.com/WordPress/wordpress-develop/pull/3080/files#diff-13e853d5729978a0e7936543d65ad1a3e120b994702c9559bf99c3ad857f5066R2431)

`if ( strpos( $css_selector, '--' ) === 0 && preg_match( '/^([a-z0-9\-]+)$/', $css_selector ) ) {`

Reply:
> Would it be an issue that `sanitize_key` forces lower case, so `--HOORAH` becomes `--hoorah`?

Thank you for the suggestion! Pushed a fix based on the code you posted 👍 

> Also `in_array( '--*', $allowed_attr, true )` is always going to be `true`, no?

Not necessarily... `$allowed_attr` is filterable, so someone may choose to disable css-vars by removing it from the array.